### PR TITLE
feat: add falcon_detection_details datasource

### PR DESCRIPTION
**Pull request by anasmuhmd**
2/https://github.com/blackstork-io/fabric/issues/243

### CODE REVIEW

File: internal/crowdstrike/data_falcon_detection_details.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+				{
+					Name: "filter",
+					Type: cty.String,
+					Doc:  "Host search expression using Falcon Query Language (FQL)",
+				},
+				{
+					Name:        "size",
+					Type:        cty.Number,
+					Constraints: constraint.Integer | constraint.RequiredNonNull,
+					Doc:         "limit the number of queried items",
```
Comment (Andrew-Morozko) [inline internal/crowdstrike/data_falcon_detection_details.go:35]: Perhaps the size should have a default value
```suggestion
					Name:        "limit",
					Type:        cty.Number,
					Constraints: constraint.Integer | constraint.NonNull,
					Doc:         "Maximum number of returned items",
					DefaultVal: cty.NumberIntVal(50),
					MinInclusive: cty.NumberIntVal(1),
```
I guess that having a default limit would be a bit neater from the UX standpoint (@traut what do you think?)

**Review by Andrew-Morozko — APPROVED** [review]
LGTM, just a tiny nitpick regarding the "size" argument

File: internal/crowdstrike/data_falcon_detection_details.go (hunk 1 above)
Comment (anasanzari) [reply internal/crowdstrike/data_falcon_detection_details.go:35]: I agree. Let me update all prs

Comment (anasmuhmd) [reply internal/crowdstrike/data_falcon_detection_details.go:35]: I agree, let me update all prs

**Pull request merged**